test/functional/email_confirmations_controller_test.rb

```diff
@@ -130,6 +130,97 @@ class EmailConfirmationsControllerTest < ActionController::TestCase
     end
   end
 
+  context "on POST to webauthn_update" do
+    context "with webauthn enabled" do
+      setup do
+        @user = create(:user)
+        @webauthn_credential = create(:webauthn_credential, user: @user)
+        get :update, params: { token: @user.confirmation_token, user_id: @user.id }
+        @challenge = session[:webauthn_authentication]["challenge"]
+        @origin = "http://localhost:3000"
+        @rp_id = URI.parse(@origin).host
+        @client = WebAuthn::FakeClient.new(@origin, encoding: false)
+        WebauthnHelpers.create_credential(
+          webauthn_credential: @webauthn_credential,
+          client: @client
+        )
```

Comment:
Since this block is repeated in the next two tests, I'd prefer it to be moved to a setup block under the `"on POST to webauthn_update"` context so it would be easier to read/organize (same with the block in passwords controller).

Reply:
This setup block slightly differs on each one. But for the stuff within the setup that's constant, I moved it higher up.